templatestore/views.py

```diff
@@ -126,13 +176,16 @@ def post_template_view(request):
 
             else:
                 template = templates[0]  # only one template should exist
-                max_version = TemplateVersion.objects.filter(
+                current_versions = TemplateVersion.objects.filter(
                     template_id=template
-                ).order_by(-Length("version"), "-version")[:1]
-                major_version, minor_version = max_version[0].version.split(".")
-                minor_version = str(int(minor_version) + 1)
+                ).values("version")
+                version_list = []
+                for version in current_versions:
+                    major, minor = version["version"].split(".")
+                    version_list.append((int(major), int(minor)))
 
-                version = major_version + "." + minor_version
+                version_list.sort(key=lambda ax: (int(ax[0]), int(ax[1])), reverse=True)
+                version = str(version_list[0][0]) + "." + str(version_list[0][1] + 1)
```

Comment:
order by id desc -> should also work 
that will give you the recent version

Reply:
yea right ...I will change it